src/pudl/output/ferc1.py

```diff
@@ -422,73 +422,53 @@ def out_ferc1__yearly_steam_plants_fuel_sched402(
-                "utility_id_pudl",
-                "utility_name_ferc1",
-                "seller_name",
-                "record_id",
-            ],
-        )
-    )
-    return purchased_power_df
+## example/draft factory pattern
+specs = [
+    {"name": "ferc1__yearly_purchased_power_and_exchanges_sched326"},
+    {"name": "ferc1__yearly_plant_in_service_sched204"},
+    {"name": "ferc1__yearly_balance_sheet_assets_sched110"},
+]
 
 
-@asset(io_manager_key="pudl_io_manager", compute_kind="Python")
-def out_ferc1__yearly_plant_in_service_sched204(
-    core_ferc1__yearly_plant_in_service_sched204: pd.DataFrame,
-    core_pudl__assn_ferc1_pudl_utilities: pd.DataFrame,
-) -> pd.DataFrame:
-    """Pull a dataframe of FERC Form 1 Electric Plant in Service data."""
-    pis_df = core_ferc1__yearly_plant_in_service_sched204.merge(
-        core_pudl__assn_ferc1_pudl_utilities, on="utility_id_ferc1"
-    ).pipe(
-        pudl.helpers.organize_cols,
-        [
-            "report_year",
-            "utility_id_ferc1",
-            "utility_id_pudl",
-            "utility_name_ferc1",
-            "record_id",
-        ],
+# draft asset_factory
+def generate_asset_factory(spec) -> AssetsDefinition:
+    """Create ferc1 assets."""
+    ins: Mapping[str, AssetIn] = {
+        f"core_{spec['name']}": AssetIn(f"core_{spec['name']}"),
+        "core_pudl__assn_ferc1_pudl_utilities": AssetIn(
+            "core_pudl__assn_ferc1_pudl_utilities"
+        ),
+    }
+
+    @asset(
+        name=f"out_{spec['name']}",
+        io_manager_key="pudl_io_manager",
+        compute_kind="Python",
+        ins=ins,
     )
-    return pis_df
+    def _asset(
+        **kwargs: dict[str, pd.DataFrame],
+    ) -> pd.DataFrame:
+        """Generate a dataframe for {} asset specification.""".format(spec["name"])
+        return_df = kwargs[f"core_{spec['name']}"].merge(
+            kwargs["core_pudl__assn_ferc1_pudl_utilities"], on="utility_id_ferc1"
+        )
+        return return_df
 
+    return _asset
 
-@asset(io_manager_key="pudl_io_manager", compute_kind="Python")
-def out_ferc1__yearly_balance_sheet_assets_sched110(
-    core_ferc1__yearly_balance_sheet_assets_sched110: pd.DataFrame,
-    core_pudl__assn_ferc1_pudl_utilities: pd.DataFrame,
-) -> pd.DataFrame:
-    """Pull a useful dataframe of FERC Form 1 balance sheet assets data."""
-    out_ferc1__yearly_balance_sheet_assets_sched110 = (
-        core_ferc1__yearly_balance_sheet_assets_sched110.merge(
-            core_pudl__assn_ferc1_pudl_utilities, on="utility_id_ferc1"
-        ).pipe(
-            pudl.helpers.organize_cols,
-            [
-                "report_year",
-                "utility_id_ferc1",
-                "utility_id_pudl",
-                "utility_name_ferc1",
-                "record_id",
-                "asset_type",
-            ],
-        )
-    )
-    return out_ferc1__yearly_balance_sheet_assets_sched110
+
+def create_generated_assets() -> list[AssetsDefinition]:
```

Comment:
Please remove this function and use a list comprehension that iterates of the table names with no prefix to call the asset factory function.